src/dpp/etf.cpp

```diff
@@ -269,28 +269,33 @@ const char* etf_parser::read_string(uint32_t length) {
 	return (const char*)str;
 }
 
-static const char* atom_null = "null";
-static const char* atom_true = "true";
-static const char* atom_false = "false";
-
 json etf_parser::process_atom(const char* atom, uint16_t length) {
 	json j;
 
 	if (atom == NULL) {
 		return j;
 	}
 
+	/**
+	 * WARNING: PERFORMANCE HOTSPOT
+	 * Valgrind cachegrind identified this as a performance hotspot in DPP when using ETF. This gets called a LOT,
+	 * and the difference here between comparing the simple constant types here where the lengths are known in this
+	 * way, compared to using memcpy or strncpy four times, is absolutely huge (factors of ten). Considering this
+	 * can be called hundreds of times a second on a busy bot, the performance of this function should not be
+	 * underestimated. Think carefully before 'tidying' this function further!
+	 * 
+	 */
 	if (length >= 3 && length <= 5) {
-		if (length == 4 && *((uint32_t*)atom) == *((uint32_t*)atom_null)) { // "null"
+		if (length == 4 && atom[0] == 'n' && atom[1] == 'u' && atom[2] == 'l' && atom[3] == 'l') { // "null"
```

Comment:
How about using `memcmp` here? Potentially could be even faster on some stdlib/compilets

Reply:
it isnt, please read comment :)

Reply:
I read it, I'm talking about `memcmp`, not `memcpy`. `memcmp` will often be optimized to type punning, except the compiler is allowed to do that